common/src/main/java/com/wynntils/mc/render/RenderUtils.java

```diff
@@ -947,6 +951,131 @@ public static void rotatePose(PoseStack poseStack, float centerX, float centerZ,
         poseStack.translate(-centerX, -centerZ, 0);
     }
 
+    public static class MapRenderer {
+        // TODO: Support circle map rendering
+        public static void renderMapQuad(
+                MapProfile map,
+                PoseStack poseStack,
+                float centerX,
+                float centerZ,
+                float textureX,
+                float textureZ,
+                float width,
+                float height,
+                float scale,
+                boolean followPlayerRotation,
+                boolean renderUsingLinear) {
+            // enable rotation if necessary
+            if (followPlayerRotation) {
+                poseStack.pushPose();
+                RenderUtils.rotatePose(
+                        poseStack, centerX, centerZ, 180 - McUtils.player().getYRot());
+            }
+
+            // has to be before setting shader texture
+            int option = renderUsingLinear ? GL11.GL_LINEAR : GL11.GL_NEAREST;
+            RenderSystem.texParameter(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_MAG_FILTER, option);
+            RenderSystem.texParameter(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_MIN_FILTER, option);
+
+            RenderSystem.setShader(GameRenderer::getPositionTexShader);
+            RenderSystem.setShaderTexture(0, map.resource());
+
+            // clamp map rendering
+            // has to be after setting shader texture
+            RenderSystem.texParameter(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_WRAP_T, GL13.GL_CLAMP_TO_BORDER);
+            RenderSystem.texParameter(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_WRAP_S, GL13.GL_CLAMP_TO_BORDER);
+
+            float uScale = 1f / map.getTextureWidth();
+            float vScale = 1f / map.getTextureHeight();
+
+            // avoid rotational overpass - This is a rather loose oversizing, if possible later
+            // use trignometry, etc. to find a better one
+            float extraFactor = 1F;
+            if (followPlayerRotation) {
+                // 1.5 > sqrt(2);
+                extraFactor = 1.5F;
+
+                if (width > height) {
+                    extraFactor *= width / height;
+                } else {
+                    extraFactor *= height / width;
+                }
+            }
+
+            float halfRenderedWidth = width / 2f * extraFactor;
+            float halfRenderedHeight = height / 2f * extraFactor;
+            float halfTextureWidth = halfRenderedWidth * scale;
+            float halfTextureHeight = halfRenderedHeight * scale;
+
+            Matrix4f matrix = poseStack.last().pose();
+
+            // TODO replace with RenderUtils after settings pr is merged
```

Comment:
Is this comment still relevant? I'm not sure what it means anyway.

Reply:
Not sure either

Reply:
Remove it then :)